app/src/main/java/nl/tudelft/b_b_w/model/GetDatabaseHandler.java

```diff
@@ -66,7 +63,7 @@ public final String getContactName(String hash) {
 
         // When returning an exception the whole program crashes,
         // but we want to preserve the state.
-        if (cursor.getCount() < 1) return null;
+        if (cursor.getCount() < 1) return "Unknown";
```

Comment:
Your PR is merged, so I'm now using the backtrack method, thus making this method redundant.